config/config.go

```diff
@@ -0,0 +1,121 @@
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/go-homedir"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// Filename into which Config is serialized
+	Filename = "config.yaml"
+
+	// DefaultConfigPath is used in the unlikely event that
+	// the user has no home directory (no ~), there is no
+	// XDG_CONFIG_HOME set
+	DefaultConfigPath = ".config/func"
+
+	// DefaultLanguage is intentionaly undefined.
+	DefaultLanguage = ""
+)
+
+type Config struct {
+	// Language Runtime
+	Language string `yaml:"language"`
+
+	// Confirm Prompts
+	Confirm bool `yaml:"confirm"`
+}
+
+// New Config struct with all members set to static defaults.  See NewDefaults
+// for one which further takes into account the optional config file.
+func New() Config {
+	return Config{
+		Language: DefaultLanguage,
+		// ...
+	}
+}
+
+// Creates a new config populated by global defaults as defined by the
+// config file located in .Path() (the global func settings path, which is
+//  usually ~/.config/func)
+func NewDefault() (cfg Config, err error) {
+	cfg = New()       // cfg now populated by static defaults
+	p := ConfigPath() // applies ~/.config/func/config.yaml if it exists
+	if _, err = os.Stat(p); err != nil {
+		if os.IsNotExist(err) {
+			err = nil // config file is not required
+		}
+		return
+	}
+	bb, err := os.ReadFile(p)
+	if err != nil {
+		return
+	}
+	err = yaml.Unmarshal(bb, &cfg) // cfg now has applied config.yaml
```

Comment:
```suggestion
    cfg, err = Load(p)
    if os.IsNotExist(err) {
		err = nil // config file is not required
	}
```

Reply:
/facepalm
Pretty sure that's the reason I added the `Load` method in the first place.
_codus interruptus_ perhaps!

Reply:
Hey @lance; while in half-sleep late last night I realized why this was the way it was.
The Config object is first instantiated with its values populated with static defaults.
Then the .yaml is loaded into that sams struct, overwriting any static defaults with user-provided global settings.
The `Load` method merely loads exactly the .yaml, ignoring static defaults.
Therefore, updating this method in this manner will ignore the `cfg` object and thus static defaults
(hence the code staticcheck complaining about the unused declaration).
This is the general path: static defaults first, user provided config values overwrite those, environment variables overwrite those, and finally flags.  👍🏻  

I refactored a bit to make this more clear so as to hopefully avoid the same confusion in the future.